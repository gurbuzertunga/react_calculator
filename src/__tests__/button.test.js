import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Button from '../components/button';

describe('checks if Button component renders correctly', () => {
  const clickHandler = () => {
    '';
  };

  test('should render Button correctly', () => {
    const tree = renderer
      .create(<Button name="" clickHandler={clickHandler} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display correct value on the screen', () => {
    render(<Button name="+" clickHandler={clickHandler} />);
    const result = screen.getByText('+');

    expect(result).toBeInTheDocument();
  });
});
