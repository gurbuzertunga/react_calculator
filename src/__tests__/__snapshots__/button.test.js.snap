// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Button component renders correctly should render Button correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  
</button>
`;
